pulp_smash/tests/rpm/api_v2/test_unassociate.py

```diff
@@ -94,6 +94,50 @@ def test_02_remaining_units(self):
         }
         self.assertEqual(removed_ids & remaining_ids, set())
 
+    def test_03_last_unit_removed(self):
+        """Assert that `last_unit_removed` timestamp was not updated.
+
+        Whenever disassociate_units is called `last_unit_removed` should not
+        be updated if that cal does not remove units.
+
+        Each verification is wrapped in a subTest.
+        See :meth:`do_verify_last_unit_timestamp`.
+        """
+        type_ids = ['erratum', 'package_category', 'package_group', 'rpm']
+        random.shuffle(type_ids)
+        for type_id in type_ids:
+            with self.subTest(type_id=type_id):
+                self.do_verify_last_unit_timestamp(type_id)
+
+    def do_verify_last_unit_timestamp(self, type_id):
+        """Verify that the `last_unit_removed` was not updated.
+
+        Do the following:
+
+        1. Note the repository's ``last_unit_removed`` field.
+        2. Sleep for at least one second.
+        3. Attempt to remove a non defined ``unit_id`` .
+        4. Note the repository's ``last_unit_removed`` field.
+
+        Verify that:
+
+        Timestamps``last_unit_removed``,  before and after the attempt
+        to unassociate an unit was not updated.
+        """
+        lur_before = self.get_repo_last_unit_removed()
+
+        time.sleep(1)  # ensure last_unit_removed increments
+        unit = random.choice(_get_units_by_type(self.initial_units, type_id))
+        self.removed_units.append(unit)
+        _unassociate_not_defined_unit(self.cfg, self.repo, unit)
```

Comment:
While defining `_unassociate_not_defined_unit` and `_get_not_defined_unit` works, it's a very verbose way of doing things. Functions are great for encapsulating code that is re-used, or that are especially logically distinct. That's not the case here. This is a simpler way to do things:

```python
unit = copy.copy(random.choice(self.initial_units))  # don't change original object
unit[_get_unit_id(unit)] = utils.uuid4()
# remove unit from repository
# capture lur_after, and assert lur_before == lur_after
```

Bam. Three lines of code instead of two functions.